crates/uv/src/main.rs

```diff
@@ -588,6 +588,20 @@ struct PipSyncArgs {
     #[clap(long, default_value_t, value_enum, env = "UV_INDEX_STRATEGY")]
     index_strategy: IndexStrategy,
 
+    /// Require a matching hash for each requirement.
+    ///
+    /// Hash-checking mode is all or nothing. If enabled, _all_ requirements must be provided
+    /// with a corresponding hash or set of hashes. Additionally, if enabled, _all_ requirements
+    /// must either be pinned to exact versions (e.g., `==1.0.0`), or be specified via direct URL.
+    ///
+    /// Hash-checking mode introduces a number of additional constraints:
+    /// - Git dependencies are not supported.
```

Comment:
Can we require a full length hash for git, or is that just a follow up PR?

Reply:
We could but pip doesn't support these so I was hesitant to deviate.